saga/order-service/src/main/java/io/debezium/examples/saga/order/saga/OrderPlacementSaga.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.examples.saga.order.saga;
+
+import static io.debezium.examples.saga.order.saga.OrderPlacementSaga.CREDIT_APPROVAL;
+import static io.debezium.examples.saga.order.saga.OrderPlacementSaga.PAYMENT;
+
+import java.util.UUID;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import io.debezium.examples.saga.framework.Saga;
+import io.debezium.examples.saga.framework.SagaBase;
+import io.debezium.examples.saga.framework.SagaStatus;
+import io.debezium.examples.saga.framework.SagaStepMessage;
+import io.debezium.examples.saga.order.event.CreditApprovalEvent;
+import io.debezium.examples.saga.order.event.PaymentEvent;
+import io.debezium.examples.saga.order.model.PurchaseOrder;
+import io.debezium.examples.saga.order.model.PurchaseOrderStatus;
+
+@Saga(type="order-placement", stepIds = {CREDIT_APPROVAL, PAYMENT})
+public class OrderPlacementSaga extends SagaBase {
+
+    protected static final String PAYMENT = "payment";
+    protected static final String CREDIT_APPROVAL = "credit-approval";
+
+    private static ObjectMapper objectMapper = new ObjectMapper();
+
+    public OrderPlacementSaga(UUID id, JsonNode payload) {
+        super(id, payload);
+    }
+
+    public static OrderPlacementSaga forPurchaseOrder(PurchaseOrder purchaseOrder) {
+        ObjectNode payload = objectMapper.createObjectNode();
+
+        payload.put("order-id", purchaseOrder.id);
+        payload.put("customer-id", purchaseOrder.customerId);
+        payload.put("payment-due", purchaseOrder.paymentDue);
+        payload.put("credit-card-no", purchaseOrder.creditCardNo);
+        payload.put("type", "REQUEST");
```

Comment:
should there be constants for payload types i.e. "REQUEST" ?

Reply:
Done.